thunx-model/src/test/java/eu/xenit/contentcloud/thunx/predicates/model/Scenarios.java

```diff
@@ -23,4 +24,59 @@ void regoPythonExample() {
         assertThat(expression.getResultType()).isEqualTo(Boolean.class);
     }
 
+    @Test
+    void mixedSimplifyableExpressionsExample() {
+        // 5 == 5 AND (FALSE OR (input.number == 1 AND TRUE) OR input.number == 3) AND input.str == "x" AND ("abc" == "abc")
+        var expression = LogicalOperation.conjunction(
+                Comparison.areEqual(Scalar.of(5), Scalar.of(5)),
+                LogicalOperation.disjunction(
+                        Scalar.of(false),
+                        LogicalOperation.disjunction(Collections.emptyList()),
+                        LogicalOperation.conjunction(
+                                Comparison.areEqual(
+                                        SymbolicReference.parse("input.number"),
+                                        Scalar.of(1)
+                                ),
+                                Scalar.of(true)
+                        ),
+                        Comparison.areEqual(
+                                SymbolicReference.parse("input.number"),
+                                Scalar.of(3)
+                        )
+                ),
+                Comparison.areEqual(
+                        SymbolicReference.parse("input.str"),
+                        Scalar.of("x")
+                ),
+                Comparison.areEqual(
+                        Scalar.of("abc"),
+                        Scalar.of("abc")
+                ),
+                LogicalOperation.conjunction(Collections.emptyList())
+        );
+
+        // This simplifies to: (input.number == 1 OR input.number == 3) AND input.str == "x"
```

Comment:
Wait, are we actually evaluating the comparison `5 == 5` and optimize it away ?

Reply:
Yes, we are. Since both sides are knowns, and the FunctionSimplifier is implemented for Operator.EQUALS, it will be evaluated. The resulting TRUE is then further removed, since a TRUE in an AND expression can be removed without affecting the expression.